core/src/components/range/range.tsx

```diff
@@ -213,12 +213,32 @@ export class Range implements ComponentInterface {
    */
   @Prop({ mutable: true }) value: RangeValue = 0;
   @Watch('value')
-  protected valueChanged() {
+  protected valueChanged(newValue: RangeValue, oldValue: RangeValue) {
+    const valuesChanged = this.areValuesDifferent(newValue, oldValue);
+    if (valuesChanged) {
+      this.ionInput.emit({ value: this.value });
+    }
+
     if (!this.noUpdate) {
       this.updateRatio();
     }
   }
 
+  /**
+   * Compares two RangeValue inputs to determine if they are different.
+   *
+   * @param newVal - The new value.
+   * @param oldVal - The old value.
+   * @returns `true` if the values are different, `false` otherwise.
+   */
+  private areValuesDifferent = (newVal: RangeValue, oldVal: RangeValue) => {
```

Comment:
```suggestion
  private compareValues = (newVal: RangeValue, oldVal: RangeValue) => {
```

Nit: can we name this something else? We don't have any other functions that begin with `are`.

Select uses a similar function here: https://github.com/ionic-team/ionic-framework/blob/9e11fcb2bb504916a1ef910a011f80b93be36109/core/src/utils/forms/compare-with-utils.ts#L10

Reply:
Done: https://github.com/ionic-team/ionic-framework/pull/30293/commits/f7ad2ba7b76fd47213f38a257dfa4c516ae3d854 and https://github.com/ionic-team/ionic-framework/pull/30293/commits/812d7674efe63983441e39e0b613a864bfcd0f56